entities/component.py

```diff
@@ -0,0 +1,199 @@
+        return 'role'
+
+
+class Required(BoolValue):
+    def __init__(self, required: bool):
+        super().__init__('required', required)
+
+
+class Negated(BoolValue):
+    def __init__(self, negated: bool):
+        super().__init__('negated', negated)
+
+
+class Clock(BoolValue):
+    def __init__(self, clock: bool):
+        super().__init__('clock', clock)
+
+
+class ForcedNet(StringValue):
+    def __init__(self, forced_net: str):
+        super().__init__('forced_net', forced_net)
+
+
+class Signal():
+    def __init__(self, uuid: str, name: Name, role: Role, required: Required,
+                 negated: Negated, clock: Clock, forced_net: ForcedNet):
+        self.uuid = uuid
+        self.name = name
+        self.role = role
+        self.required = required
+        self.negated = negated
+        self.clock = clock
+        self.forced_net = forced_net
+
+    def __str__(self) -> str:
+        ret = '(signal {} {} {}\n'.format(self.uuid, self.name, self.role) +\
+            ' {} {} {} {}\n'.format(self.required, self.negated, self.clock, self.forced_net) +\
+            ')'
+        return ret
+
+
+class SymbolUUID(UUIDValue):
+    def __init__(self, symbol_uuid: str):
+        super().__init__('symbol', symbol_uuid)
+
+
+class SignalUUID(UUIDValue):
+    def __init__(self, signal_uuid: str):
+        super().__init__('signal', signal_uuid)
+
+
+class TextDesignator(EnumValue):
+    SYMBOL_PIN_NAME = 'pin'
+    SIGNAL_NAME = 'signal'
+
+    def get_name(self) -> str:
+        return 'text'
+
+
+class PinSignalMap():
+    def __init__(self, pin_uuid: str, signal_uuid: SignalUUID,
+                 text_designator: TextDesignator):
+        self.pin_uuid = pin_uuid
+        self.signal_uuid = signal_uuid
+        self.text_designator = text_designator
+
+    def __str__(self) -> str:
+        return '(pin {} {} {})'.format(self.pin_uuid, self.signal_uuid, self.text_designator)
+
+
+class Suffix(StringValue):
+    def __init__(self, suffix: str):
+        super().__init__('suffix', suffix)
+
+
+class Gate():
+    def __init__(self, uuid: str, symbol_uuid: SymbolUUID, position: Position,
+                 rotation: Rotation, required: Required, suffix: Suffix):
+        self.uuid = uuid
+        self.symbol_uuid = symbol_uuid
+        self.position = position
+        self.rotation = rotation
+        self.required = required
+        self.suffix = suffix
+        self.pins = []  # type: List[PinSignalMap]
+
+    def add_pin_signal_map(self, pin_signal_map: PinSignalMap) -> None:
+        self.pins.append(pin_signal_map)
+
+    def __str__(self) -> str:
+        ret = '(gate {}\n'.format(self.uuid) +\
+            ' {}\n'.format(self.symbol_uuid) +\
+            ' {} {} {} {}\n'.format(self.position, self.rotation, self.required, self.suffix)
+        for pin in self.pins:
+            ret += ' {}\n'.format(pin)
+        ret += ')'
+        return ret
+
+
+class Norm(EnumValue):
+    EMPTY = '""'
+    # TODO not sure about those
```

Comment:
What's the open question here?

Reply:
What the string value of the other possible values is. Would need to check in the file.

Reply:
Fixed.